src/bindings/python/src/openvino/runtime/opset13/ops.py

```diff
@@ -255,8 +257,13 @@ def scaled_dot_product_attention(
 
     :return: The new node performing Scaled Dot Product Attention operation.
     """
-    inputs = as_nodes(query, key, value, attention_mask) if attention_mask is not None else as_nodes(
-        query, key, value, scale)
+    inputs = as_nodes(query, key, value)
+    if attention_mask is not None:
+        inputs.append(as_node(attention_mask))
+    elif scale is not None:
+        inputs.append(as_node(convert_like(constant(np.array(0, np.int32)), inputs[0])))
+    if scale is not None:
+        inputs.append(as_node(scale))
```

Comment:
What is the difference between setting the mask to 0 when scale is not None and having mask = None when scale is None?

Reply:
If I understand your question correctly, it is just for user experience to allow to use optional inputs like in C++ API. So when:

- both attn and scale is None, it will just pass query, key and value inputs to C++ constructor; 
- attn is provided and scale is None - pass query, key, value and attention;
- attn is None and scale is provided - fill attn with const 0 for C++ constructor and pass query, key, value, attn, scale;
- both attn and scale provided - pass all inputs